src/cow-react/modules/advancedOrders/hooks/useParseNumberOfParts.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from 'react'
+import { useSetAtom } from 'jotai'
+import {
+  updateAdvancedOrdersSettingsAtom,
+  NumberOfPartsError,
+} from '@cow/modules/advancedOrders/state/advancedOrdersSettingsAtom'
+import { MIN_PARTS_NUMBER, MAX_PARTS_NUMBER } from '@src/constants'
+
+export function useParseNumberOfParts() {
+  const updateSettingsState = useSetAtom(updateAdvancedOrdersSettingsAtom)
+
+  return useCallback(
+    (value: string) => {
+      const parsed = parseInt(value) || 0
```

Comment:
I think this is the root cause of Elena's comment on the `0` parts.
Should we default to `1` instead of `0` to no create an invalid state?

Reply:
yes, at 1 part

Reply:
however, here i think might be fine to set the default to zero, if this is what he is handling

![image](https://github.com/cowprotocol/cowswap/assets/2352112/96b012b1-b501-4d0d-a0f8-4c1c35783e7a)